### PR TITLE
Clear Search Results

### PR DESCRIPTION
**Pull request by F3PiX**
#379 
Add link to clear search results, disabled until there is a search submitted.
- [x] add link to clear current Search results
- [x] add styling to link
- [ ] ~~hide link until search results on display~~

### CODE REVIEW

**Comment by F3PiX** [comment]
@katrin-k The stage is yours ;-)

**Comment by katrin-k** [comment]
@F3PiX I just wanted to take care of the link, but it doesn't render as a link when I run it locally. 
The output misses the HTML:
```
<form …>
  <input name="search" …>
  <input name="commit" …>
  Clear search results
</form>
```

Did some code snippet play lemming?

Re: the 3rd list item in this PR (hide the link): I'd rather put this into a separate + new issue to not delay the merge of this PR for too long. (Plus, I'm not 100% convinced this is really necessary given the Ajax effort…)

Re: the X icon: I'd rather not use it this way. From a UX pattern/user perspective, the X inside an input field first and foremost removes the entry done by the user, not the result. Usually, the result is then reloaded (more or less) seamlessly. Which again brings us to Ajax… And since we reload the page I'd go for a standard button. Wdyt?

**Comment by F3PiX** [comment]
Haha, that got me confused. Turns out, I wrote this in html.erb. I am going to slim it up right now.

**Comment by F3PiX** [comment]
Totally agree with everything you say @katrin-k  
I hope the .erb file was the problem, couldn't reproduce it myself.

**Comment by katrin-k** [comment]
@F3PiX strangely, .erb was not the problem. Slim also doesn't render the HTML here… :crying_cat_face: 

Could anyone else take a look, plz? 
Tl;dr On my machine, `link_to_if` does not render the HTML. Works fine for Maud, though.

File: app/views/users/_search.html.slim [hunk 1]
```diff
@@ -0,0 +1,4 @@
+= form_tag users_path, method: :get, id: "search-form" do |f|
+  = text_field_tag :search, params[:search], placeholder: "Search by name"
+  = submit_tag "Go", class: "btn btn-default"
+  = link_to_if(params[:search], "Clear search results", users_path)
```
Comment (carpodaster) [inline app/views/users/_search.html.slim]: `params[:search].present?` would be better in case the param got carried over as `…?search=`, i.e. w/o a value that could be interpreted as an empty string. An empty string is still a truthy value in Ruby.

**Comment by F3PiX** [comment]
This is taking html escaping to a whole new level.

**Comment by carpodaster** [comment]
@F3PiX is this still WIP? Looks good to merge.

That does the `==` do in Slim?

**Comment by F3PiX** [comment]
Good morning karpermeester. It is still WIP, working with katrin_k on the link.
== is output without html escaping. Jury is still out on this solution.

**Comment by katrin-k** [comment]
Upsi, I just realized I made a mistake in describing the problem. When there's no search param, the link text is displayed without any HTML tags (but it's displayed, which it shouldn't). The link properly renders when a search param is there. `params[:search].present?` doesn't make a difference. 

What does the trick is defining an else. 
```
= link_to_if(params[:search].present?, "Clear search results", users_path, class: "btn btn-default")  do
 a
end
```

However, the `end` is then rendered as HTML tags: `<end></end>`. Removing the `end` helps in the way that there's only an empty a link tag when there's no search param, but I'll keep it since I have a hunch this is better Ruby syntax and thus can't interfere with anything else. 

@F3PiX I'll also add the `params[:search].present?` to my commit.

**Comment by katrin-k** [comment]
Close due to duplicate in #511.
